app/uk/gov/hmrc/lisaapi/controllers/BulkPaymentController.scala

```diff
@@ -87,29 +108,31 @@ class BulkPaymentController @Inject()(
     )
   }
 
-  private def withValidDates(startDate: String, endDate: String)
+  private def withValidDates(startDate: Option[String], endDate: Option[String])
                             (success: (DateTime, DateTime) => Future[Result])
                             (implicit startTime: Long): Future[Result] = {
-
-    val start = parseDate(startDate)
-    val end = parseDate(endDate)
-
-    (start, end) match {
-      case (Some(s), Some(e)) => {
-        withDatesWithinBusinessRules(s, e) { () =>
-          success(s, e)
-        }
+    val dates = for {
+      start <- startDate
+      end <- endDate
+    } yield (start, end)
+
+    dates.map(dates => {
+      val parseStart = parseDate(dates._1)
+      val parseEnd = parseDate(dates._2)
+      (parseDate(dates._1), parseDate(dates._2)) match {
+        case (Some(s), Some(e)) =>
+          withDatesWithinBusinessRules(s, e)(() => success(s, e))
+        case (None, Some(_)) =>
+          lisaMetrics.incrementMetrics(startTime, BAD_REQUEST, LisaMetricKeys.TRANSACTION)
+          Future.successful(BadRequest(Json.toJson(ErrorBadRequestStart)))
+        case (Some(_), None) =>
+          lisaMetrics.incrementMetrics(startTime, BAD_REQUEST, LisaMetricKeys.TRANSACTION)
+          Future.successful(BadRequest(Json.toJson(ErrorBadRequestEnd)))
+        case _ =>
+          lisaMetrics.incrementMetrics(startTime, BAD_REQUEST, LisaMetricKeys.TRANSACTION)
+          Future.successful(BadRequest(Json.toJson(ErrorBadRequestStartEnd)))
       }
-      case (None, Some(_)) =>
-        lisaMetrics.incrementMetrics(startTime, BAD_REQUEST, LisaMetricKeys.TRANSACTION)
-        Future.successful(BadRequest(Json.toJson(ErrorBadRequestStart)))
-      case (Some(_), None) =>
-        lisaMetrics.incrementMetrics(startTime, BAD_REQUEST, LisaMetricKeys.TRANSACTION)
-        Future.successful(BadRequest(Json.toJson(ErrorBadRequestEnd)))
-      case _ =>
-        lisaMetrics.incrementMetrics(startTime, BAD_REQUEST, LisaMetricKeys.TRANSACTION)
-        Future.successful(BadRequest(Json.toJson(ErrorBadRequestStartEnd)))
-    }
+    }).getOrElse(Future.successful(MethodNotAllowed(Json.toJson(ErrorMethodNotAllowed))))
```

Comment:
I think bad request would be more appropriate here? I've usually only seen method not allowed used to signify an incorrect HTTP verb (i.e. POST on a GET url)